### ISSUE TITLE
Electrode explorer in docker problems

### ISSUE THREAD

**Issue by LeeFrancis** [electrode-io/electrode-explorer#8]
When I build and deploy to Docker and when I hit 
  http://localhost:3000/api/update/REPO/component
I get an error :

Error encountered: ENOENT: no such file or directory, open '/usr/src/app/data/REPO/component.json'

as a response.

I added a bit of logging and I see that update handlers fetchRepo is trying to read :
/usr/src/app/data/REPO/component.json and is complaining because it does not exist.
I dump out my meta data from the github call and that looks complete.

The message after these logs are 
module info stderr npm info it worked if it ends with ok
 npm info using npm@3.10.9
 npm info using node@v4.6.2
 npm info attempt registry request try #1 at 7:00:44 PM
 npm http request GET https://registry.npmjs.org/react-intl
 npm http 304 https://registry.npmjs.org/react-intl
 npm info ok 

Pretty sure my env variables are correct and that all my github tokens and npmrc file is there.
Also, this works fine running on my local machine not in a Docker container.

Any ideas?

Thanks,

Lee.

**Comment by caoyangs** [comment]
Thank you for checking out explorer. I was unable to reproduce the issue.

The file `/usr/src/app/data/REPO/component.json` should have been created here https://github.com/electrode-io/electrode-explorer/blob/master/server/component-api/check-dependencies.js#L89

which is invoked by method `fetchRepo`
https://github.com/electrode-io/electrode-explorer/blob/example/server/component-api/update-handler.js#L76
https://github.com/electrode-io/electrode-explorer/blob/example/server/component-api/fetch-repo.js#L56

Basically the file should exist after `fetchRepo` call, can you check from there why the file wasn't created?

**Comment by LeeFrancis** [comment]
Looks like your latest commit fixed my problem.
https://github.com/electrode-io/electrode-explorer/commit/57fd73addfc8d85e87d2c80997b66d93d62a8586

Thanks for the great work, this whole project is awesome.

**Comment by LeeFrancis** [comment]
OK, my mistake..
Seems like it is still failing.
I see the following in the docker container logs :

```
error getting module dependencies [Error: npm info it worked if it ends with ok
npm info using npm@3.10.9
npm info using node@v4.5.0
npm info attempt registry request try #1 at 11:10:16 PM
npm http request GET https://registry.npmjs.org/gulp
npm http 200 https://registry.npmjs.org/gulp
npm info ok 
]
```

But, when I attach to the running container and run : 

/usr/src/app/scripts/info-module.sh gulp

it works just fine.

Running docker on mac - Docker version 1.12.3, build 6b644ec

This is my Dockerfile 
`FROM node:4.5
RUN npm i -g npm@3
RUN apt-get update && apt-get install -y --no-install-recommends cron
RUN chmod 644 /etc/crontab
EXPOSE 3000
ENV DIR /usr/src/app
RUN mkdir -p $DIR
WORKDIR $DIR
COPY . $DIR
COPY .npmrc /root
RUN npm install
RUN $DIR/node_modules/.bin/gulp build
ENV GHACCESS_TOKEN xxxxxxxxxxxxxxxxxx
ENV NODE_ENV production
CMD node server`

Sorry to reopen.. Still a great project!

**Comment by LeeFrancis** [comment]
As a follow up I commented out the stderr check ignoring the whole block and it worked.

https://github.com/electrode-io/electrode-explorer/blob/master/server/component-api/check-dependencies.js#L22

**Comment by caoyangs** [comment]
Glad to know you like it.

You may want to check why `npm info` fails, does it have right access? But anyway this shouldn't be blocking, can you check out this commit https://github.com/electrode-io/electrode-explorer/commit/be57d16718511e4cba4ca65ed7ece71b56c7f1e2 It should let you go through despite npm error.

**Comment by LeeFrancis** [comment]
Yes.. That worked. Not sure why ``npm info`` fails, it works fine when I attach to the docker image and run by hand.
Thanks for this fix.
